core/domain/exp_services_test.py

```diff
@@ -3057,3 +3057,148 @@ def test_reject_actioned_suggestion(self):
+
+    def test_draft_version_valid_true(self):
+        self.assertTrue(exp_services.is_draft_version_valid(
+            self.EXP_ID1, self.USER_ID))
+
+    def test_draft_version_valid_false(self):
+        self.assertFalse(exp_services.is_draft_version_valid(
+            self.EXP_ID2, self.USER_ID))
+
+    def test_create_or_update_draft_older_draft_exists(self):
+        with self.swap(exp_domain.Exploration, 'validate',
+                       self._empty_method):
+            exp_services.create_or_update_draft(
+                self.EXP_ID1, self.USER_ID, self.NEW_CHANGELIST, 5,
+                self.NEWER_DATETIME)
+        exp_user_data = user_models.ExplorationUserDataModel.get(
+            self.USER_ID, self.EXP_ID1)
+        self.assertEqual(exp_user_data.exploration_id, self.EXP_ID1)
+        self.assertEqual(
+            exp_user_data.draft_change_list, self.NEW_CHANGELIST)
+        self.assertEqual(exp_user_data.draft_change_list_last_updated,
+                         self.NEWER_DATETIME)
+        self.assertEqual(exp_user_data.draft_change_list_exp_version, 5)
+
+    def test_create_or_update_draft_newer_draft_exists(self):
+        exp_services.create_or_update_draft(
+            self.EXP_ID1, self.USER_ID, self.NEW_CHANGELIST, 5,
+            self.OLDER_DATETIME)
+        user_exp = user_models.ExplorationUserDataModel.get(
+            self.USER_ID, self.EXP_ID1)
+        self.assertEqual(user_exp.exploration_id, self.EXP_ID1)
+        self.assertEqual(user_exp.draft_change_list, self.draft_change_list)
+        self.assertEqual(
+            user_exp.draft_change_list_last_updated, self.DATETIME)
+        self.assertEqual(user_exp.draft_change_list_exp_version, 2)
+
+    def test_create_or_update_draft_draft_does_not_exist(self):
+        with self.swap(exp_services, 'apply_change_list',
+                       self._get_exp_domain_object):
+            with self.swap(exp_domain.Exploration, 'validate',
+                           self._empty_method):
+                exp_services.create_or_update_draft(
+                    self.EXP_ID3, self.USER_ID, self.NEW_CHANGELIST, 5,
+                    self.NEWER_DATETIME)
+        exp_user_data = user_models.ExplorationUserDataModel.get(
+            self.USER_ID, self.EXP_ID3)
+        self.assertEqual(exp_user_data.exploration_id, self.EXP_ID3)
+        self.assertEqual(
+            exp_user_data.draft_change_list, self.NEW_CHANGELIST)
+        self.assertEqual(exp_user_data.draft_change_list_last_updated,
+                         self.NEWER_DATETIME)
+        self.assertEqual(exp_user_data.draft_change_list_exp_version, 5)
+
+    def test_get_exp_with_draft_applied_draft_exists(self):
+        apply_change_list_counter = test_utils.CallCounter(
+            self._get_exp_domain_object)
+        with self.swap(exp_services, 'apply_change_list',
+                       apply_change_list_counter):
+            exp_services.get_exp_with_draft_applied(
```

Comment:
If I understand correctly you're asking me to compare  exploration.init_state.param_changes to self.param_changes. I don't think that will work. That's because while we are applying the change list to the exploration, we aren't then saving it to the backend. Just returning the resulting domain object.

I guess what we can instead do is capture the params passed while calling apply_change_list and assert if they are correct.(Not sure if this is what you meant in the first place)